internal/dependency/kv_get_test.go

```diff
@@ -167,16 +140,181 @@ func TestNewKVGetQuery(t *testing.T) {
 
 	for i, tc := range cases {
 		t.Run(fmt.Sprintf("%d_%s", i, tc.name), func(t *testing.T) {
-			act, err := NewKVExistsQuery(tc.i)
-			if (err != nil) != tc.err {
-				t.Fatal(err)
+			act, err := NewKVGetQuery(tc.i)
+
+			if act != nil {
+				act.stopCh = nil
 			}
 
+			if tc.err {
+				assert.Error(t, err)
+			} else {
+				exp := &KVGetQuery{KVExistsQuery: *tc.exp}
+				assert.Equal(t, exp, act)
+			}
+		})
+	}
+}
+
+func TestNewKVGetQueryV1(t *testing.T) {
```

Comment:
🤦‍♀️  I added this comment to the wrong thread:

> Just wanted to follow up with where I'm at! Another issue I ran into was that the input format is different between the two methods, where one takes a string with the options included and the other takes a string for the key and options as a slice. Still trying to figure out a way to make it work since we'll be running into this code duplication every time we add a new V1 method.

But anyway, I ended up splitting the cases of options vs. no options so that the no options test cases could be consolidated! The git diff isn't great to look at, but it's mostly splitting that test and then having a test helper method for the common assertions.

Reply:
No worries :) I like the test consolidation a lot. I find the split between options and no-options is very thoughtful and readable - it makes it very clear how the functions are different. Also find the reusable verifyNewKV*Query() cleaned up the testing nicely